### PR TITLE
feat: Added Schema type

### PR DESCRIPTION
**Pull request by gismya**
- [ ] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [x] The description contains manual test instructions

## Changes

Added a type for schema and made related code changes. Made a change in getPrimaryKeyAttributes to give null if the Primary Key array would be empty.

## Test

Are schemas still working as expected?

### CODE REVIEW

File: source/session.ts [hunk 1]
```diff
@@ -722,10 +724,10 @@ export class Session {
    * @param  {string} schemaId Id of schema model, e.g. `AssetVersion`.
    * @return {Object|null} Schema definition
    */
-  getSchema(schemaId: string): Data | null {
+  getSchema(schemaId: string): Schema | null {
     for (const index in this.schemas) {
-      if (this.schemas[index].id === schemaId) {
-        return this.schemas[index];
+      if (this.schemas[Number(index)].id === schemaId) {
+        return this.schemas[Number(index)];
```
Comment (jimmycallin) [inline source/session.ts:730]: consider rewriting this to use this.schemas.find instead

File: source/session.ts [hunk 2]
```diff
@@ -705,7 +707,7 @@ export class Session {
           entityType,
           primaryKeys.map((key: string) => updateEntity[key]),
           Object.keys(data).reduce<T>((accumulator, key: keyof T) => {
-            if (primaryKeys.indexOf(key) === -1) {
+            if (primaryKeys.indexOf(key.toString()) === -1) {
```
Comment (jimmycallin) [inline source/session.ts:710]: is this really necessary? can't we do something like keyof T extends String?

**Review by jimmycallin — CHANGES_REQUESTED** [review]

File: source/session.ts (hunk 2 above)
Comment (gismya) [reply source/session.ts:710]: We can type cast key as a string, but then I think this is more proper

File: source/session.ts (hunk 1 above)
Comment (gismya) [reply source/session.ts]: Done

File: source/session.ts [hunk 3]
```diff
@@ -656,7 +658,7 @@ export class Session {
       );
     }
 
-    const primaryKeys = this.getPrimaryKeyAttributes(entityType);
+    const primaryKeys = this.getPrimaryKeyAttributes(entityType) ?? [];
```
Comment (jimmycallin) [inline source/session.ts:661]: if primaryKeys returns an empty object, the resulting expression is not a valid expression. it is better to simply throw (or assert that getPrimaryKeyAttributes always returns a non-empty array)

**Review by jimmycallin — CHANGES_REQUESTED** [review]

**Review by jimmycallin — APPROVED** [review]

**Pull request merged**